src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -147,7 +151,8 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event): vo
                     if (\is_array($payload) && !empty($constraints) && !$constraints instanceof Assert\All) {
                         $constraints = new Assert\All($constraints);
                     }
-                    $violations->addAll($this->validator->validate($payload, $constraints, $argument->validationGroups ?? null));
+                    $groups = $this->resolveValidationGroups($argument->validationGroups ?? null, $event);
```

Comment:
Is the call to `resolveValidationGroups()` needed if there are no groups? Maybe this would be better:

```suggestion
                    if ($groups = $argument->validationGroups ?? null) {
                        $groups = $this->resolveValidationGroups($groups, $event);
                    }
```